### PR TITLE
[Fix](group commit) Fix cloud group commit be select  strategy

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

In #35558, we optimized be select for group commit. However, we forgot to apply this strategy to cloud. This PR applys it.

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 1]
```diff
@@ -433,9 +433,29 @@ private TNetworkAddress selectLocalRedirectBackend(boolean groupCommit, HttpServ
...
+    private TNetworkAddress selectCloudRedirectBackend(String clusterName, HttpServletRequest req, boolean groupCommit,
+            long tableId)
             throws LoadException {
-        Backend backend = StreamLoadHandler.selectBackend(clusterName, groupCommit);
+        Backend backend = null;
+        if (groupCommit) {
+            ConnectContext ctx = new ConnectContext();
+            ctx.setEnv(Env.getCurrentEnv());
+            ctx.setThreadLocalInfo();
+            ctx.setRemoteIP(req.getRemoteAddr());
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:444]: should consider clouster

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 2]
```diff
@@ -433,9 +433,29 @@ private TNetworkAddress selectLocalRedirectBackend(boolean groupCommit, HttpServ
...
+            // We set this variable to fulfill required field 'user' in
+            // TMasterOpRequest(FrontendService.thrift)
+            ctx.setQualifiedUser(Auth.ADMIN_USER);
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:447]: this is generally the same as selectLocalRedirectBackend, it's better to abstract it

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 3]
```diff
@@ -647,4 +651,29 @@ private String getAllHeaders(HttpServletRequest request) {
...
+        if (isCloud) {
+            ctx.setCloudCluster(clusterName);
+        }
+
+        Backend backend = null;
+        try {
+            backend = Env.getCurrentEnv().getGroupCommitManager()
+                    .selectBackendForGroupCommit(tableId, ctx, isCloud);
+        } catch (DdlException e) {
+            throw new RuntimeException(e);
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:675]: throw DdlException directly, RuntimeException is strange

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 4]
```diff
@@ -102,21 +102,25 @@ public Object load(HttpServletRequest request, HttpServletResponse response,
...
+                } catch (Exception e) {
+                    LOG.info("exception:" + e);
+                    return new RestBaseResult(e.getMessage());
                 }
-            } catch (Exception e) {
-                LOG.info("exception:" + e);
-                return new RestBaseResult(e.getMessage());
+            } else if (groupCommitStr.equalsIgnoreCase("sync_mode")) {
+                groupCommit = true;
             }
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:124]: throw exception if group commit is set but none of async_mode or sync_mode

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 5]
```diff
@@ -147,21 +151,29 @@ public Object streamLoadWithSql(HttpServletRequest request, HttpServletResponse
...
+                    }
+                } catch (Exception e) {
+                    LOG.info("exception:" + e);
+                    return new RestBaseResult(e.getMessage());
                 }
-            } catch (Exception e) {
-                LOG.info("exception:" + e);
-                return new RestBaseResult(e.getMessage());
+            } else if (groupCommitStr.equalsIgnoreCase("sync_mode")) {
+                groupCommit = true;
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:176]: table_id is -1 for sync_mode

**Comment by Yukang-Lian** [comment]
run buildall

**Review by mymeiyi — APPROVED** [review]

**Comment by Yukang-Lian** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**